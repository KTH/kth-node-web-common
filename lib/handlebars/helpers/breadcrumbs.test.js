const handlebars = require('handlebars')
const log = require('@kth/log')
const { registerBreadcrumbHelper, breadcrumbs, getBreadcrumbsMarkup } = require('./breadcrumbs')

jest.mock('@kth/log')
jest.mock('handlebars')

describe('registerBreadcrumbHelper', () => {
  it('Registers breadcrumbs helper', () => {
    registerBreadcrumbHelper()
    expect(handlebars.registerHelper).toHaveBeenCalledWith('breadcrumbs', breadcrumbs)
  })
})

describe('getBreadcrumbsMarkup', () => {
  it('should return null and log a warning if the input is not a valid path list', () => {
    const invalidPathList = [{}]
    const result = breadcrumbs(invalidPathList, 'en')
    expect(result).toBeNull()
    expect(log.warn).toHaveBeenCalledWith(
      '[breadcrumbs] helper requires first parameter to be a list of path item objects'
    )
  })

  it('should generate breadcrumbs HTML for a valid path list in English', () => {
    const validPathList = [
      { url: '/', label: 'Kth' },
      { url: '/test1', label: 'Test 1' },
      { url: '/test2', label: 'Test 2' },
    ]
    const result = getBreadcrumbsMarkup(validPathList, 'en')
    expect(result).toMatchSnapshot()
  })

  it('should generate breadcrumbs HTML for a valid path list in Swedish', () => {
    const validPathList = [
      { url: '/', label: 'Kth' },
      { url: '/test1', label: 'Test 1' },
      { url: '/test2', label: 'Test 2' },
    ]
    const result = getBreadcrumbsMarkup(validPathList, 'sv')
    expect(result).toMatchSnapshot()
  })
})
