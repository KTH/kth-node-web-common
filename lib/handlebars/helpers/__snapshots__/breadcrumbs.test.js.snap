// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBreadcrumbsMarkup should generate breadcrumbs HTML for a valid path list in English 1`] = `
"
  <nav id="breadcrumbs" aria-label="Breadcrumbs" class="kth-breadcrumbs">
    <ol class="kth-breadcrumbs__list">
      <li><a href="/">Kth</a></li><li><a href="/test1">Test 1</a></li><li><a href="/test2">Test 2</a></li>
    </ol>
  </nav>
"
`;

exports[`getBreadcrumbsMarkup should generate breadcrumbs HTML for a valid path list in Swedish 1`] = `
"
  <nav id="breadcrumbs" aria-label="BrÃ¶dsmulor" class="kth-breadcrumbs">
    <ol class="kth-breadcrumbs__list">
      <li><a href="/">Kth</a></li><li><a href="/test1">Test 1</a></li><li><a href="/test2">Test 2</a></li>
    </ol>
  </nav>
"
`;
